plugins/modules/vmware_vcenter_settings_info.py

```diff
@@ -0,0 +1,224 @@
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+module: vmware_vcenter_settings_info
+short_description: Gather info vCenter settings
+description:
+  - This module can be used to gather information about vCenter settings.
+author:
+  - sky-joker (@sky-joker)
+requirements:
+  - python >= 2.7
+  - PyVmomi
+extends_documentation_fragment:
+  - community.vmware.vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: "Gather info about vCenter settings"
+  community.vmware.vmware_vcenter_settings_info:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: no
+  register: vcenter_settings_info
+'''
+
+RETURN = r'''
+vcenter_config_info:
+  description: dict of vCenter settings
+  returned: success
+  type: dict
+  sample: |
+    {
+        "db_event_cleanup_previous": true,
+        "db_event_retention_previous": 30,
+        "db_max_connections_previous": 50,
+        "db_task_cleanup_previous": true,
+        "db_task_retention_previous": 30,
+        "directory_query_limit_previous": true,
+        "directory_query_limit_size_previous": 5000,
+        "directory_timeout_previous": 60,
+        "directory_validation_period_previous": 1440,
+        "directory_validation_previous": true,
+        "logging_options_previous": "info",
+        "mail_sender_previous": "",
+        "mail_server_previous": "",
+        "runtime_managed_address_previous": "",
+        "runtime_server_name_previous": "vcenter.local",
+        "runtime_unique_id_previous": 48,
+        "snmp_1_community_previous": "public",
+        "snmp_1_enabled_previous": true,
+        "snmp_1_url_previous": "localhost",
+        "snmp_2_community_previous": "",
+        "snmp_2_enabled_previous": false,
+        "snmp_2_url_previous": "",
+        "snmp_3_community_previous": "",
+        "snmp_3_enabled_previous": false,
+        "snmp_3_url_previous": "",
+        "snmp_4_community_previous": "",
+        "snmp_4_enabled_previous": false,
+        "snmp_4_url_previous": "",
+        "snmp_receiver_1_port_previous": 162,
+        "snmp_receiver_2_port_previous": 162,
+        "snmp_receiver_3_port_previous": 162,
+        "snmp_receiver_4_port_previous": 162,
+        "timeout_long_operations_previous": 120,
+        "timeout_normal_operations_previous": 30
+    }
+'''
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible_collections.community.vmware.plugins.module_utils.vmware import PyVmomi, vmware_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+
+
+class VmwareVcenterSettingsInfo(PyVmomi):
+    def __init__(self, module):
+        super(VmwareVcenterSettingsInfo, self).__init__(module)
+
+        if not self.is_vcenter():
+            self.module.fail_json(msg="You have to connect to a vCenter server!")
+
+    def ensure(self):
+        result = {}
+        option_manager = self.content.setting
```

Comment:
@sky-joker Do you think we can add something like - https://github.com/ansible-collections/vmware/blob/d8da2a5ec707d8540bac740e1aa7785c1d00a19e/plugins/modules/vmware_datacenter_info.py#L31

Reply:
I've added a schema and properties option.  
How is this?